lib/internal/source_map/source_map_cache.js

```diff
@@ -205,14 +203,26 @@ function dataFromUrl(sourceURL, sourceMappingURL) {
 // from. This allows translation from byte offset V8 coverage reports,
 // to line/column offset Source Map V3.
 function lineLengths(content) {
-  // We purposefully keep \r as part of the line-length calculation, in
-  // cases where there is a \r\n separator, so that this can be taken into
-  // account in coverage calculations.
-  return ArrayPrototypeMap(RegExpPrototypeSymbolSplit(/\n|\u2028|\u2029/, content), (line) => {
-    return line.length;
-  });
+  const contentLength = content.length;
+  const output = [];
+  let lineLength = 0;
+  for (let i = 0; i < contentLength; i++, lineLength++) {
+    const codePoint = content.codePointAt(i);
+
+    // We purposefully keep \r as part of the line-length calculation, in
+    // cases where there is a \r\n separator, so that this can be taken into
+    // account in coverage calculations.
+    // codepoints for \n, \u2028 and \u2029
+    if (codePoint === 10 || codePoint === 0x2028 || codePoint === 0x2029) {
+      output.push(lineLength);
+      lineLength = -1; // To not count the matched codePoint such as \n character
+    }
+  }
+  output.push(lineLength);
```

Comment:
```suggestion
  ArrayPrototypePush(output, lineLength);
```

Reply:
ArrayPrototypePush has performance implications according to `primordials.md`. Since adding primordials is optional, I'm strongly against adding it for `array.prototype.push` in this PR.

Reply:
I think for source map support, we don't want user code to be able to mess with internal implementation – i.e. we should take the potential performance penalty and generate reliably correct stack traces. Also, we should measure the perf difference, as it might be negligable.